File: CheckoutExampleObjC/CheckoutExampleObjC/CheckoutViewController.m
#import "CheckoutViewController.h"
@import KountDataCollector;
@implementation CheckoutViewController

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
    
    // Create a unique Session ID
    CFUUIDRef uuidRef = CFUUIDCreate(nil);
    CFStringRef uuidStrRef = CFUUIDCreateString(nil, uuidRef);
    CFRelease(uuidRef);
    NSString *sessionID = [(__bridge NSString *)uuidStrRef stringByReplacingOccurrencesOfString:@"-" withString:@""];
    CFRelease(uuidStrRef);
    self.textView.text = @"Collection Starting\n\n";
    self.textView.text = [self.textView.text stringByAppendingFormat:@"Session ID:\n%@\n\n", sessionID];
    [[KDataCollector sharedCollector] collectForSession:sessionID completion:^(NSString * _Nonnull sessionID, BOOL success, NSError * _Nullable error) {
        if (success) {
            self.textView.text = [self.textView.text stringByAppendingString:@"Collection Successful"];
            NSLog(@"Collection Status %@", [KountAnalyticsViewController getKDataCollectionStatus]);
        }
        else {
            self.textView.text = [self.textView.text stringByAppendingString:@"Collection Failed\n\n"];
            self.textView.text = [self.textView.text stringByAppendingString:[error description]];
            NSLog(@"Collection Status %@", [KountAnalyticsViewController getKDataCollectionError]);
        }
    }];
}

- (IBAction)done:(id)sender 
{
    [self.delegate checkoutViewControllerDidFinish:self];
}

@end

File: CheckoutExampleSwift/CheckoutExampleSwift/CheckoutViewController.swift
import UIKit

protocol CheckoutViewControllerDelegate: class {
    func didFinish(_ sender: CheckoutViewController)
}

class CheckoutViewController: KountAnalyticsViewController {
    
    @IBOutlet weak var textView: UITextView?
    weak var delegate:CheckoutViewControllerDelegate?
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        var sessionID = UUID().uuidString
        sessionID = sessionID.replacingOccurrences(of: "-", with: "")
        textView!.text = "Collection Starting\n\n"
        textView!.text = textView!.text.appendingFormat("Session ID:\n%@\n\n", sessionID)
        KDataCollector.shared().collect(forSession: sessionID) { (sessionID, success, error) in
            if success {
                self.textView!.text = self.textView!.text + "Collection Successful"
                print(KountAnalyticsViewController.getKDataCollectionStatus() as Any)
            } else {
                self.textView!.text = self.textView!.text + "Collection Failed\n\n"
                if ((error) != nil) {
                  self.textView!.text = self.textView!.text + error!.localizedDescription
                    print(KountAnalyticsViewController.getKDataCollectionError() as Any)
                }
            }
        }
        
    }
    
    @IBAction func done(_ sender: AnyObject) {
        self.delegate?.didFinish(self)
    }
}
